junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/NestedWithInheritanceTests.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2015-2020 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.jupiter.engine;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+class NestedWithInheritanceTests extends SuperClass {
+
+	static List<String> lifecycleInvokingClassNames;
+
+	static String OUTER = NestedWithInheritanceTests.class.getSimpleName();
+	static String NESTED = NestedClass.class.getSimpleName();
+	static String NESTEDNESTED = NestedClass.NestedNestedClass.class.getSimpleName();
+
+	@Nested
+	class NestedClass extends SuperClass {
```

Comment:
Did you verify that these nested test classes are actually executed within the Gradle build?

I'm asking, because they don't match the `*Tests` naming pattern.

Reply:
For such types of tests, we often write a single "normal" test class that uses the `EngineTestKit` or `Launcher` API to verify that the particular use cases are actually executed. So you might want to switch to that style of testing for these.

Reply:
> Did you verify that these nested test classes are actually executed within the Gradle build?
> 
> I'm asking, because they don't match the `*Tests` naming pattern.

The inner tests are indeed executed by the gradle build because it picks up the outermost test.

Reply:
Yes, I am aware of, and have used, both options: explicit integration tests and EngineTestKit-based tests.
I am inclined to leave this one in the present style as the necessary verification is achieved and changing it to TestKit would require extra effort.